sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -137,8 +137,8 @@ abstract class PartitioningAwareFileIndex(
   }
 
   /** Returns the list of files that will be read when scanning this relation. */
-  override def inputFiles: Array[SparkPath] =
-    allFiles().map(SparkPath.fromFileStatus).toArray
+  override def inputFiles: Array[String] =
+    allFiles().map(fs => SparkPath.fromFileStatus(fs).urlEncoded).toArray
```

Comment:
So, this part is not reverting to the original code, `allFiles().map(_.getPath.toUri.toString).toArray`, right?

Reply:
Ah good point. It is not exactly a revert, but is reusing the SparkPath code to translate from hadoop `FileStatus` to url-encoded string.